modules/programs/zsh/prezto.nix

```diff
@@ -423,12 +424,30 @@ in
   config = lib.mkIf cfg.enable (
     lib.mkMerge [
       {
-        home.file."${relToDotDir ".zprofile"}".source = "${cfg.package}/share/zsh-prezto/runcoms/zprofile";
-        home.file."${relToDotDir ".zlogin"}".source = "${cfg.package}/share/zsh-prezto/runcoms/zlogin";
-        home.file."${relToDotDir ".zlogout"}".source = "${cfg.package}/share/zsh-prezto/runcoms/zlogout";
         home.packages = [ cfg.package ];
 
-        home.file."${relToDotDir ".zshenv"}".source = "${cfg.package}/share/zsh-prezto/runcoms/zshenv";
+        home.file."${relToDotDir ".zprofile"}".text = ''
+          # Generated by Nix
+          . ${cfg.package}/share/zsh-prezto/runcoms/zprofile
+          ${cfgZsh.profileExtra}
```

Comment:
Changing from `.source` to `.text` and using source to load the `zsh-prezto` files so we can avoid doing a IFD: https://github.com/nix-community/home-manager/pull/6905.

Reply:
BTW, https://github.com/nix-community/home-manager/pull/6905 is very likely the reason we had the second issue I described in the PR (the environment variables not loading anymore).